pkg/impersonation/impersonation.go

```diff
@@ -34,17 +35,17 @@ func (i *Impersonator) SetUpImpersonation() (*corev1.ServiceAccount, error) {
 	if err != nil {
 		return nil, err
 	}
+	role, err := i.checkAndCreateRole()
```

Comment:
It's weird how this breaks in-between error handling of errors during checking for SA and returning SA if it already exists. Could we move check/creation of role before we check for SA for more logical flow? Maybe there is a reason it has to be this way, but I'm failing to see it since `sa` is not used here.

Reply:
I refactored this to use the role as the existence check so we don't need to have this double resource check.